src/buckets/components/context/lineProtocol.tsx

```diff
@@ -0,0 +1,137 @@
+// Libraries
+import React, {FC, useCallback, useState} from 'react'
+import {useSelector} from 'react-redux'
+
+// Utils
+import {postWrite} from 'src/client'
+import {getErrorMessage} from 'src/utils/api'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+
+// Types
+import {RemoteDataState, LineProtocolTab, WritePrecision} from 'src/types'
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface LineProtocolContextType {
+  body: string
+  handleResetLineProtocol: () => void
+  handleSetBody: (_: string) => void
+  handleSetTab: (tab: LineProtocolTab) => void
+  precision: WritePrecision
+  setPrecision: (_: WritePrecision) => void
+  tab: LineProtocolTab
+  writeError: string
+  writeLineProtocol: (_: string) => void
+  writeStatus: RemoteDataState
+}
+
+export const DEFAULT_CONTEXT: LineProtocolContextType = {
+  body: '',
+  handleResetLineProtocol: () => {},
+  handleSetBody: (_: string) => {},
+  handleSetTab: (_: LineProtocolTab) => {},
+  precision: WritePrecision.Ns,
+  setPrecision: (_: WritePrecision) => {},
+  tab: 'Upload File',
+  writeError: '',
+  writeLineProtocol: (_: string) => {},
+  writeStatus: RemoteDataState.NotStarted,
+}
+
+export const LineProtocolContext = React.createContext<LineProtocolContextType>(
+  DEFAULT_CONTEXT
+)
+
+export const LineProtocolProvider: FC<Props> = React.memo(({children}) => {
+  const [body, setBody] = useState('')
+  const [tab, setTab] = useState<LineProtocolTab>('Upload File')
+  const [precision, setPrecision] = useState(WritePrecision.Ns)
+  const [writeStatus, setWriteStatus] = useState(RemoteDataState.NotStarted)
+  const [writeError, setWriteError] = useState('')
+
+  const org = useSelector(getOrg).name
+
+  const handleResetLineProtocol = () => {
+    setBody('')
+    setWriteStatus(RemoteDataState.NotStarted)
+    setWriteError('')
+  }
+
+  const writeLineProtocol = useCallback(
+    async (bucket: string) => {
+      try {
+        setWriteStatus(RemoteDataState.Loading)
+        const resp = await postWrite({
+          data: body,
+          query: {org, bucket, precision},
+        })
+
+        if (resp.status === 204) {
```

Comment:
That's not a bad idea. Not sure how we'd enforce that / refactor it but I like keeping it tied together. I'm wondering if we should have it set to a type that's generated through the yml, since that's where the original API status codes are defined.

Reply:
Are we doing anything to change the status codes beyond their meaning in RFC 7231?  I'm sure there's plenty of places that have the defaults as constants but I see we're using 429 differently below and the more we diverge from the standards the harder this is to track.